src/main/java/frc/robot/util/ShuffleboardUtil.java

```diff
@@ -0,0 +1,160 @@
+package frc.robot.util;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.revrobotics.SparkMaxPIDController;
+
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.networktables.NetworkTableValue;
+
+public class ShuffleboardUtil {
+    /**
+     * Polls a Shuffleboard entry for a `double` change, updating the provided SparkMax PID controller's P gain if it is found.
+     * @param entry The shuffleboard entry to poll.
+     * @param pidController The PID controller to update.
+     */
+    public static void pollShuffleboardP(GenericEntry entry, SparkMaxPIDController pidController) {
+        pollShuffleboardValue(entry, (value) -> pidController.setP(value.getDouble()));
```

Comment:
perhaps we can pass in a lambda to run if the value is changed instead of manually defining all these methods?